[CI] CloseWhileRelocatingShardsIT.testCloseWhileRelocatingShards failures

See https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+internalClusterTest/173/console for an example failure

It looks to be a timing issue, it doesn't reproduce for me:
```
./gradlew :server:integTest \
  -Dtests.seed=B04B0599D97CB9C5 \
  -Dtests.class=org.elasticsearch.indices.state.CloseWhileRelocatingShardsIT \
  -Dtests.method="testCloseWhileRelocatingShards" \
  -Dtests.security.manager=true \
  -Dtests.locale=ro \
  -Dtests.timezone=Europe/London \
  -Dcompiler.java=11 \
  -Druntime.java=8
```

```
2> WARNING: Uncaught exception in thread: Thread[elasticsearch[node_sd2][generic][T#4],5,TGRP-CloseWhileRelocatingShardsIT]
  2> java.lang.AssertionError: java.lang.InterruptedException
  2> 	at __randomizedtesting.SeedInfo.seed([B04B0599D97CB9C5]:0)
  2> 	at org.elasticsearch.indices.state.CloseWhileRelocatingShardsIT.lambda$testCloseWhileRelocatingShards$1(CloseWhileRelocatingShardsIT.java:159)
  2> 	at org.elasticsearch.test.transport.StubbableTransport$WrappedConnection.sendRequest(StubbableTransport.java:223)
  2> 	at org.elasticsearch.transport.TransportService.sendRequestInternal(TransportService.java:626)
  2> 	at org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:541)
  2> 	at org.elasticsearch.transport.TransportService.submitRequest(TransportService.java:503)
  2> 	at org.elasticsearch.transport.TransportService.submitRequest(TransportService.java:494)
  2> 	at org.elasticsearch.indices.recovery.PeerRecoveryTargetService.lambda$doRecovery$3(PeerRecoveryTargetService.java:263)
  2> 	at org.elasticsearch.common.util.CancellableThreads.executeIO(CancellableThreads.java:108)
  2> 	at org.elasticsearch.indices.recovery.PeerRecoveryTargetService.doRecovery(PeerRecoveryTargetService.java:257)
  2> 	at org.elasticsearch.indices.recovery.PeerRecoveryTargetService.access$500(PeerRecoveryTargetService.java:84)
  2> 	at org.elasticsearch.indices.recovery.PeerRecoveryTargetService$RecoveryRunner.doRun(PeerRecoveryTargetService.java:658)
  2> 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:751)
  2> 	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
  2> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
  2> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
  2> 	at java.lang.Thread.run(Thread.java:748)
  2> Caused by: java.lang.InterruptedException
  2> 	at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:998)
  2> 	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304)
  2> 	at java.util.concurrent.CountDownLatch.await(CountDownLatch.java:231)
  2> 	at org.elasticsearch.indices.state.CloseWhileRelocatingShardsIT.lambda$testCloseWhileRelocatingShards$1(CloseWhileRelocatingShardsIT.java:156)
  2> 	... 15 more
```

--- a/server/src/test/java/org/elasticsearch/indices/state/CloseWhileRelocatingShardsIT.java
+++ b/server/src/test/java/org/elasticsearch/indices/state/CloseWhileRelocatingShardsIT.java
@@ -119,6 +119,8 @@ public class CloseWhileRelocatingShardsIT extends ESIntegTestCase {
 
         final String targetNode = internalCluster().startDataOnlyNode();
         ensureClusterSizeConsistency(); // wait for the master to finish processing join.
+        final MockTransportService targetTransportService =
+            (MockTransportService) internalCluster().getInstance(TransportService.class, targetNode);
 
         final Set<String> acknowledgedCloses = ConcurrentCollections.newConcurrentSet();
         try {
@@ -146,8 +148,7 @@ public class CloseWhileRelocatingShardsIT extends ESIntegTestCase {
                 }
 
                 final DiscoveryNode sourceNode = clusterService.state().nodes().resolveNode(primary.currentNodeId());
-                ((MockTransportService) internalCluster().getInstance(TransportService.class, targetNode))
-                    .addSendBehavior(internalCluster().getInstance(TransportService.class, sourceNode.getName()),
+                targetTransportService.addSendBehavior(internalCluster().getInstance(TransportService.class, sourceNode.getName()),
                         (connection, requestId, action, request, options) -> {
                             if (PeerRecoverySourceService.Actions.START_RECOVERY.equals(action)) {
                                 logger.debug("blocking recovery of shard {}", ((StartRecoveryRequest) request).shardId());
@@ -210,28 +211,30 @@ public class CloseWhileRelocatingShardsIT extends ESIntegTestCase {
                     }
                 }
             }
+
+            for (String index : indices) {
+                if (acknowledgedCloses.contains(index)) {
+                    assertIndexIsClosed(index);
+                } else {
+                    assertIndexIsOpened(index);
+                }
+            }
+
+            targetTransportService.clearAllRules();
+
+            assertThat("Consider that the test failed if no indices were successfully closed", acknowledgedCloses.size(), greaterThan(0));
+            assertAcked(client().admin().indices().prepareOpen("index-*"));
+            ensureGreen(indices);
+
+            for (String index : acknowledgedCloses) {
+                long docsCount = client().prepareSearch(index).setSize(0).get().getHits().getTotalHits().value;
+                assertEquals("Expected " + docsPerIndex.get(index) + " docs in index " + index + " but got " + docsCount
+                    + " (close acknowledged=" + acknowledgedCloses.contains(index) + ")", (long) docsPerIndex.get(index), docsCount);
+            }
         } finally {
             assertAcked(client().admin().cluster().prepareUpdateSettings()
                 .setTransientSettings(Settings.builder()
                     .putNull(EnableAllocationDecider.CLUSTER_ROUTING_REBALANCE_ENABLE_SETTING.getKey())));
-        }
-
-        for (String index : indices) {
-            if (acknowledgedCloses.contains(index)) {
-                assertIndexIsClosed(index);
-            } else {
-                assertIndexIsOpened(index);
-            }
-        }
-
-        assertThat("Consider that the test failed if no indices were successfully closed", acknowledgedCloses.size(), greaterThan(0));
-        assertAcked(client().admin().indices().prepareOpen("index-*"));
-        ensureGreen(indices);
-
-        for (String index : acknowledgedCloses) {
-            long docsCount = client().prepareSearch(index).setSize(0).get().getHits().getTotalHits().value;
-            assertEquals("Expected " + docsPerIndex.get(index) + " docs in index " + index + " but got " + docsCount
-                + " (close acknowledged=" + acknowledgedCloses.contains(index) + ")", (long) docsPerIndex.get(index), docsCount);
         }
     }
 }

